File: 项目文件/frontend/src/types/project-meeting.ts
/** 项目会议/交流记录类型定义 */

export interface ProjectMeeting {
  id: string;
  project_id: string;
  number: string;
  type: string;
  started_at: string;
  speaker: string | null;
  participants: string[];
  content: string | null;
  notes: unknown[];
  created_at: string;
  updated_at: string;
}

export interface ProjectMeetingCreate {
  project_id: string;
  number: string;
  type: string;
  started_at: string;
  speaker?: string;
  participants?: string[];
  content?: string;
  notes?: unknown[];
}

export interface ProjectMeetingUpdate {
  number?: string;
  type?: string;
  started_at?: string;
  speaker?: string;
  participants?: string[];
  content?: string;
  notes?: unknown[];
}

export interface ProjectMeetingListResponse {
  items: ProjectMeeting[];
  total: number;
}

File: 项目文件/backend/app/schemas/project_meeting.py
"""项目会议 Pydantic 模型"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class ProjectMeetingCreate(BaseModel):
    """创建项目会议请求"""

    project_id: uuid.UUID
    number: str
    type: str
    started_at: datetime
    speaker: str | None = None
    participants: list[str] = []
    content: str | None = None
    notes: list[Any] = []


class ProjectMeetingUpdate(BaseModel):
    """更新项目会议请求"""

    number: str | None = None
    type: str | None = None
    started_at: datetime | None = None
    speaker: str | None = None
    participants: list[str] | None = None
    content: str | None = None
    notes: list[Any] | None = None


class ProjectMeetingResponse(BaseModel):
    """项目会议响应"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    number: str
    type: str
    started_at: datetime
    speaker: str | None = None
    participants: list[Any] = []
    content: str | None = None
    notes: list[Any] = []
    created_at: datetime
    updated_at: datetime


class ProjectMeetingListResponse(BaseModel):
    """项目会议列表响应"""

    items: list[ProjectMeetingResponse]
    total: int
